smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/element/UserTuneElement.java

```diff
@@ -0,0 +1,258 @@
+        private Integer track;
+        private URI uri;
+
+        private Builder() {
+        }
+
+        /**
+         * Artist is an optional element in UserTuneElement.
+         * @param artist
+         * @return Builder
+         */
+        public Builder setArtist(String artist) {
+            this.artist = artist;
+            return this;
+        }
+
+        /**
+         * Length is an optional element in UserTuneElement.
+         * @param length
+         * @return Builder
+         */
+        public Builder setLength(int length) {
+            if (length > 0) {
+                this.length = length;
+                return this;
+            }
+            else {
+                throw new IllegalArgumentException("Length should be positive. In case of live streaming, length is expected to be null");
+            }
+        }
+
+        /**
+         * Rating is an optional element in UserTuneElement.
+         * The rating to be given is suppossed to be out of 10.
+         * <br>
+         * Method will throw an Exception if rating is less than 0 or greater than 10.
+         * @param rating
+         * @return Builder
+         */
+        public Builder setRating(int rating) {
+            if (rating <= 10 && rating >= 0) {
+                this.rating = rating;
+            } else {
+                throw new IllegalArgumentException("Rating tried to set was out of bounds");
+            }
+            return this;
+        }
+
+        /**
+         * Source is an optional element in UserTuneElement.
+         * @param source
+         * @return Builder
+         */
+        public Builder setSource(String source) {
+            this.source = source;
+            return this;
+        }
+
+        /**
+         * Title is a compulsory element in UserTuneElement.
+         * @param title
+         * @return Builder
+         */
+        public Builder setTitle(String title) {
+            this.title = title;
+            return this;
+        }
+
+        /**
+         * Track is an optional element in UserTuneElement.
+         * @param track
+         * @return Builder
+         */
+        public Builder setTrack(int track) {
+            if (track > 0) {
+                this.track = track;
+                return this;
+            }
+            else {
+                throw new IllegalArgumentException("Track number should be positive.");
+            }
+        }
+
+        public Builder setUri(URI uri) {
+            this.uri = uri;
+            return this;
+        }
+
+        /**
+         * This method is called to build a UserTuneElement.
+         * @throws IllegalArgumentException
+         * @return UserTuneElement
+         */
+        public UserTuneElement build() {
+            if (title != null) {
+                return new UserTuneElement(this);
+            }
+            else {
+                throw new IllegalArgumentException("UserTune should atleast have 'title'");
```

Comment:
Same simplification to improve the readability of the method as above.

Besides I don't see a requirement in XEP-0118 that user tune element's must have at least 'title' set (although it appears sensible). Did I miss it?

Reply:
No, it is not stated anywhere. Though not sharing `title` would make less sense, I understand that it is not wise to contravene a XEP. I will make the corresponding changes.